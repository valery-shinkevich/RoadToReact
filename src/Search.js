import React from 'react'

const Search = ({value,onChange,children}) => 
<form>
    <span>{children}</span>
    <input type="text" value={value} onChange={onChange} />
</form>;

export default Search