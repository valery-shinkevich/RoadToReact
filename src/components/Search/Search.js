import React from "react"

const Search = ({ value, hitsPerPage, onChange, onHppChange, onSubmit, children }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={value} onChange={onChange} />
    Строк на странице:
    <input style={{width: 20}} type="text" value={hitsPerPage} onChange={onHppChange} />
    <button type="submit">{children}</button>
  </form>
)

export default Search