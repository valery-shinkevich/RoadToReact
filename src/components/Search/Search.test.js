import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Search  from './Search'

describe('Search', () => {

  it('отрисовывает без ошибки', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search>Поиск</Search>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('есть корректный снимок', () => {
    const component = renderer.create(
      <Search>Поиск</Search>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
