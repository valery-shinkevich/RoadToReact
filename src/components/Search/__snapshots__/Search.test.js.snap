// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search есть корректный снимок 1`] = `
<form>
  <input
    type="text"
  />
  Строк на странице:
  <input
    style={
      Object {
        "width": 20,
      }
    }
    type="text"
  />
  <button
    type="submit"
  >
    Поиск
  </button>
</form>
`;
